packages/vue/src/components/molecules/SfModal/SfModal.js

```diff
@@ -0,0 +1,78 @@
+export default {
+  name: "SfModal",
+  props: {
+    /**
+     * Cross closing modal button
+     */
+    cross: {
+      type: Boolean,
+      default: true
+    },
+    /**
+     * Visibility of the modal
+     */
+    visible: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * Whether to show the overlay
+     */
+    overlay: {
+      type: Boolean,
+      default: true
+    },
+    /**
+     * If true clicking outside will not dismiss the modal
+     */
+    persistent: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * overlay transition effect
+     */
+    transitionOverlay: {
+      type: String,
+      default: "fade"
+    },
+    /**
+     * overlay transition effect
+     */
+    transitionModal: {
+      type: String,
+      default: "fade"
+    }
+  },
+  mounted() {
+    const keydownHandler = e => {
+      if (e.key === "Escape" || e.key === "Esc" || e.keyCode === 27) {
+        this.$emit("close");
+      }
+    };
+    document.addEventListener("keydown", keydownHandler);
+    this.$once("hook:destroyed", () => {
+      document.removeEventListener("keydown", keydownHandler);
+    });
+  },
+  methods: {
+    checkPersistence() {
+      if (this.persistent === false) {
+        this.$emit("close");
+      }
+    }
```

Comment:
to handle `visible` prop as you're doing (and i think it's necessary to abstract the transitions), i think you should have a `toggle` method, this should emit close/open events but also set toggle `visible` value, and add/remove the keydown listener

Reply:
then you would not only call `$emit("close")`, but call `toggle()`, which would emit the event and do the other necessary stuff

Reply:
you mean this only for the click on the overlay? but why should I use a toggle method if the click on the overlay will always close the modal and never open it? 🤔 

the overlay behaves the same as the close button but with a check in addition.
this.persistent check is only to handle the event close on it or not (that's never close the modal if clicking on the overlay)

Reply:
hey @lorenaramonda, sorry for the delay.
not only for the overlay, I'm suggesting to use `toggle` on every places you're closing or opening the modal, you can also create two functions, `open` and `close`, It's up to you...

Reply:
i just think using `toggle` instead of `open` and `close` you'll write less code

Reply:
if you prefer, you can create a `open` method, called watching `visible` when it's changed from false to true, and you create also a `close` method, called when close button is clicked and when overlay is clicked (if persistent isn't true)

Reply:
then `open` would emit open event and add the event listener to DOM, and `close` would set visible to false, remove the event listener and emit close event